test/field_exchanger_tests.jl

```diff
@@ -308,4 +312,72 @@ for FT in (Float32, Float64)
     @testset "step_model_sims! for FT=$FT" begin
         @test FieldExchanger.step_model_sims!((; stub = TestSurfaceSimulation1(FT(0))), 1) === nothing
     end
+
+    @testset "exchange! for FT=$FT" begin
+        # Here we exchange two albedo fields from the surface to the atmos,
+        #  and two precipitation fields from the atmos to the surface.
+        # coupler cache setup
+        boundary_space = CC.CommonSpaces.CubedSphereSpace(FT; radius = FT(6371e3), n_quad_points = 4, h_elem = 4)
+        coupler_field_names =
+            (:surface_direct_albedo, :surface_diffuse_albedo, :P_liq, :P_snow, :ρ_sfc, :T_sfc, :F_radiative, :temp1)
+        coupler_fields = NamedTuple{coupler_field_names}(
+            ntuple(i -> CC.Fields.ones(boundary_space) .* FT(0.5), length(coupler_field_names)),
+        )
+
+        # model cache setup
+        atmos_names =
+            (:albedo_direct, :albedo_diffuse, :liquid_precipitation, :snow_precipitation, :radiative_energy_flux_sfc)
+        atmos_fields = NamedTuple{atmos_names}(ntuple(i -> CC.Fields.ones(boundary_space), length(atmos_names)))
```

Comment:
This fails because you are constructing a NamedTuple of Fields instead of a Field of NamedTuples.